### PR TITLE
Edit venue

### PR DESCRIPTION
**Pull request by Jessica-Mach**
This is the edit venue user story. It adds the following: 

- When user goes to "/admin/concert-venues, user sees the list of concert venues with an edit button. This edit button does not appear on "/concert-venues".
- Admin is currently set to true automatically when user visits the admin site 
- Logging in as admin will be a separate user story
- Clicking edit button takes the user to an edit form
- Changing anything in the form will update the venue in the database
- Successful update redirects to the venue show page (which has not been built yet)
- I checked that the update persisted and did not duplicate in the database

### CODE REVIEW

File: src/main/frontend/components/EditVenue.js [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+      const responseBody = await response.json();
+      setFormPayload(responseBody.venue);
+    } catch (error) {
+      console.error(`There was an error in fetch: ${error}`);
+    }
+  };
+
+  const editVenue = async () => {
+    try {
+      const response = await fetch(`/api/v1/admin/${venueId}`, {
```
Comment (safang99) [inline src/main/frontend/components/EditVenue.js:46]: I might update the path to something like `api/v1/admin/venues/${venueId}`.  In the future if there are other resources an admin can edit we will want them namespaced appropriately.

File: src/main/frontend/components/EditVenue.js [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+  });
+
+  const [errors, setErrors] = useState({});
+  const [shouldRedirect, setShouldRedirect] = useState(false);
+  const [venueId, setVenueId] = useState(null);
+
+  const fetchVenue = async () => {
+    try {
+      const id = props.match.params.id;
+      setVenueId(id);
```
Comment (safang99) [inline src/main/frontend/components/EditVenue.js:30]: While this shouldn't cause any issues we can avoid updating state at multiple levels in the same function and just use  `props.match.params.id` in the PUT fetch or define this outside of the fetch function so it is available to both functions.

File: src/main/frontend/components/EditVenue.js [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+          const body = await response.json();
+          return setErrors(body.errors);
+        } else {
+          const errorMessage = `${response.status} (${response.statusText})`;
+          const error = new Error(errorMessage);
+          throw error;
+        }
+      }
+      const body = await response.json();
+      console.log("body", body);
```
Comment (safang99) [inline src/main/frontend/components/EditVenue.js:64]: We can remove 64 now that it has been tested and is working

File: src/main/frontend/components/EditVenue.js [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+      const id = props.match.params.id;
+      setVenueId(id);
+      const response = await fetch(`/api/v1/venues/${id}`);
+      if (!response.ok) {
+        const errorMessage = `${response.status} (${response.statusText})`;
+        const error = new Error(errorMessage);
+        throw error;
+      }
+      const responseBody = await response.json();
+      setFormPayload(responseBody.venue);
```
Comment (safang99) [inline src/main/frontend/components/EditVenue.js:38]: It looks like the `responseBody.venue` will also return with the correct `id` if you want to reference that

File: src/main/frontend/components/EditVenue.js [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+      address: "",
+      city: "",
+      state: "",
+      zipCode: "",
+      capacity: "",
+      description: "",
+      imgUrl: "",
+      phoneNumber: ""
+    });
+    setErrors({});
```
Comment (safang99) [inline src/main/frontend/components/EditVenue.js:100]: Nice 🙌

**Review by safang99 — APPROVED** [review]
Looks awesome! The naming issue is more something to think about but not a change you have to make.  I will leave that up to you as a team to decide.  After making a decision feel free to go ahead and merge the changes 👍

File: src/main/frontend/components/EditVenue.js (hunk 1 above)
Comment (Jessica-Mach) [reply src/main/frontend/components/EditVenue.js:46]: Agreed. I updated this as suggested

File: src/main/frontend/components/EditVenue.js (hunk 2 above)
Comment (Jessica-Mach) [reply src/main/frontend/components/EditVenue.js:30]: I updated to define the id outside the fetch and removed the venueId state.

File: src/main/frontend/components/EditVenue.js (hunk 4 above)
Comment (Jessica-Mach) [reply src/main/frontend/components/EditVenue.js:38]: Noted. Since I removed the venueID state, this is no longer needed.

**Pull request merged**